### ISSUE TITLE
Window control overlay does not dim when modal custom dialog shows

### ISSUE THREAD

**Issue by bpasero** [microsoft/vscode#159671]
When you open a folder on a fresh user data dir, you see a modal workspace trust custom dialog that dims the background but not the new WCO:

<img width="773" alt="image" src="https://user-images.githubusercontent.com/900690/187691646-b4518f45-76fe-4f92-8151-1b4b82453523.png">

**Comment by deepak1556** [comment]
Nice catch! colors can for the overlay can be controlled with https://github.com/electron/electron/blob/main/docs/api/browser-window.md#winsettitlebaroverlayoptions-windows.

**Comment by bpasero** [comment]
Yeah, its a bit of a layer break dance, we probably would have to call that method from here:

https://github.com/microsoft/vscode/blob/2ccc23bef5bfb823df79f60c2cc6b06fe24de805/src/vs/workbench/browser/parts/dialogs/dialogHandler.ts#L85

When a custom dialog opens, but since the layer is wrong we need a specific dialog handler for custom dialogs in `electron-sandbox`.

Note that there is even a user setting to enable custom dialogs for all dialogs on desktop.

**Comment by bpasero** [comment]
Actually it could easily be added where we process dialogs from the desktop app. We know here if a custom or native dialog opens:

https://github.com/microsoft/vscode/blob/2335d959257a86e290bd3e693298d387f3000c6e/src/vs/workbench/electron-sandbox/parts/dialogs/dialog.contribution.ts#L64

We even have access to the native host service, the only challenge is to find the right color for the WCO. We seem to be adding a large `div` over the workbench with `rgba(0, 0, 0, 0.3)` and would have to compute the new title background and foreground color that this brings.

**Comment by bpasero** [comment]
If someone wants to tackle this, opening for help. Note, when you run out of sources, you have to force enable WCO here:

https://github.com/microsoft/vscode/blob/2335d959257a86e290bd3e693298d387f3000c6e/src/vs/platform/window/common/window.ts#L167

And some pseudo code to update the WCO is:

```ts
this.nativeHostService.updateWindowControls({
	backgroundColor: withNullAsUndefined(this.getColor(TITLE_BAR_ACTIVE_BACKGROUND, color => color.blend(new Color(new RGBA(0, 0, 0, 0.3))))),
	foregroundColor: withNullAsUndefined(this.getColor(TITLE_BAR_ACTIVE_FOREGROUND, color => color.blend(new Color(new RGBA(0, 0, 0, 0.3))))
});
```

**Comment by najmiehsa** [comment]
I work on this one.

**Comment by sbatten** [comment]
Just throwing another perspective here: the current non-dimmed controls are an accurate representation of the fact that these buttons are still functional, unlike the rest of the title bar.

**Comment by yume-chan** [comment]
![image](https://user-images.githubusercontent.com/1330321/196238034-02be4796-6da9-40a6-952c-5a48e36c4c61.png)

Windows Terminal will dim window control buttons when popups are shown (they are still functional). I only know it's now Win32 caption buttons, but not sure whether it's from WinUI or Windows Terminal itself.

Traditionally, when model dialogs are shown, the window behind can't be interacted. A recent exception is web browsers, but that's due to being abused by some malicious websites. I personally don't mind if they can still work when dimmed, but I prefer the dimming to provide a more consistent look, and make users focus more on the model dialog current shown.
